sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriter.cs

```diff
@@ -195,7 +223,10 @@ internal static bool ShouldWriteAsJson(IPersistableModel<object> model, ModelRea
 
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     private static bool IsJsonFormatRequested<T>(IPersistableModel<T> model, ModelReaderWriterOptions options)
-        => options.Format == "J" || (options.Format == "W" && model.GetFormatFromOptions(options) == "J");
+    {
+        var format = options.TryGetProxy(out IPersistableModel<T>? proxy) ? proxy.GetFormatFromOptions(options) : model.GetFormatFromOptions(options);
```

Comment:
Curious: can the proxy specify a different format from the base model?  It might be helpful to draw out that scenario, maybe in tests?